src/Uno.UI/UI/Xaml/Controls/DropDownButton/DropDownButton.cs

```diff
@@ -1,89 +1,83 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+// MUX reference DropDownButton.cpp, tag winui3/release/1.4.2
+
 using Uno.UI.Helpers.WinUI;
 using Windows.UI.Xaml.Automation;
 using Windows.UI.Xaml.Automation.Peers;
 
-namespace Windows.UI.Xaml.Controls
-{
-	public partial class DropDownButton : Button
-	{
-		private bool m_isFlyoutOpen;
+namespace Windows.UI.Xaml.Controls;
 
-		public DropDownButton()
-		{
-			DefaultStyleKey = typeof(DropDownButton);
-		}
+public partial class DropDownButton : Button
+{
+	private bool m_isFlyoutOpen;
 
-		protected override AutomationPeer OnCreateAutomationPeer()
-		{
-			return new DropDownButtonAutomationPeer(this);
-		}
+	public DropDownButton()
+	{
+		DefaultStyleKey = typeof(DropDownButton);
+	}
 
-		protected override void OnApplyTemplate()
-		{
-			base.OnApplyTemplate();
+	protected override void OnApplyTemplate()
+	{
+		base.OnApplyTemplate();
 
-			RegisterFlyoutEvents();
-		}
+		RegisterFlyoutEvents();
+	}
 
-		internal override void OnPropertyChanged2(DependencyPropertyChangedEventArgs args)
+	private void RegisterFlyoutEvents()
+	{
+		if (Flyout != null)
 		{
-			if (args.Property == Button.FlyoutProperty)
-			{
-				OnFlyoutPropertyChanged(this, args);
-			}
-
-			base.OnPropertyChanged2(args);
+			Flyout.Opened += OnFlyoutOpened;
+			Flyout.Closed += OnFlyoutClosed;
 		}
+	}
 
-		private void RegisterFlyoutEvents()
-		{
-			if (Flyout != null)
-			{
-				Flyout.Opened += OnFlyoutOpened;
-				Flyout.Closed += OnFlyoutClosed;
-			}
-		}
+	internal bool IsFlyoutOpen()
+	{
+		return m_isFlyoutOpen;
+	}
 
-		internal bool IsFlyoutOpen()
+	internal void OpenFlyout()
+	{
+		if (Flyout is Flyout)
 		{
-			return m_isFlyoutOpen;
+			Flyout.ShowAt(this);
 		}
+	}
 
-		internal void OpenFlyout()
+	internal void CloseFlyout()
+	{
+		if (Flyout is Flyout)
```

Comment:
Use `if (Flyout is { } flyout)` and then `flyout.Hide()` on the next line (same above for `ShowAt`